File: verminmud/src/org/vermin/wicca/web/Output.js
/**
 * WICCA Output component
 * Copyright (c) Vermin ry, 2005
 */

var Output = {
	
  put: function(txt) {
    //alert(txt);
    txt = new String(txt);
    var tv = this.getTextView();
    tv.append(txt);
  },
  
  showLink: function(title, url) {
    window.open(url, title);
  },

  getTextView: function() {
    if(!("textview" in this))
      this.textview = new TextView("game-output");
    return this.textview;
  }
  

};

wicca[2] = [Output, null, "put", "showLink"];

File: verminmud/src/org/vermin/wicca/web/Output.java
package org.vermin.wicca.web;

public class Output extends AbstractWebComponent implements
		org.vermin.wicca.remote.Output {

	public Output(WebClientOutput wco) { super(wco); }
	
	public void put(String msg) {
		write((short)2,(byte)1, msg);
	}
	public void showLink(String title, String url) {
		write((short)2, (byte)2, title, url);
	}
}
